torchmdnet/models/tensornet.py

```diff
@@ -236,16 +257,22 @@ def forward(
             edge_vec is not None
         ), "Distance module did not return directional information"
         # Distance module returns -1 for non-existing edges, to avoid having to resize the tensors when we want to ensure static shapes (for CUDA graphs) we make all non-existing edges pertain to a ghost atom
-        # Total charge q is a molecule-wise property. We transform it into an atom-wise property, with all atoms belonging to the same molecule being assigned the same charge q
-        if q is None:
-            q = torch.zeros_like(z, device=z.device, dtype=z.dtype)
-        else:
-            q = q[batch]
         zp = z
+        if extra_args is not None:
+            # we are assuming that extra args will be used, see model.py forward method how extra_args is passed
+            for label, t in extra_args.items():
+                # molecule wise --> atom wise 
+                if t.shape != z.shape:
+                    extra_args[label] = t[batch]
+            
         if self.static_shapes:
             mask = (edge_index[0] < 0).unsqueeze(0).expand_as(edge_index)
             zp = torch.cat((z, torch.zeros(1, device=z.device, dtype=z.dtype)), dim=0)
-            q = torch.cat((q, torch.zeros(1, device=q.device, dtype=q.dtype)), dim=0)
+            if self.additional_labels is not None:
+                for label in self.additional_methods.keys():
+                    assert label in extra_args, f"Extra field {label} not found in extra_args"
```

Comment:
This assert should go where my previous comment is

Reply:
Yes, I will also move the `static_shapes` assertion and reshaping above, so the for loop on the additional_labels will only take place once